src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -961,7 +963,9 @@ private void NoteWrite(BoundExpression n, BoundExpression value, bool read)
                             if ((object)_sourceAssembly != null)
                             {
                                 var field = fieldAccess.FieldSymbol.OriginalDefinition;
-                                _sourceAssembly.NoteFieldAccess(field, read: value == null || WriteConsideredUse(fieldAccess.FieldSymbol.Type, value), write: true);
+                                _sourceAssembly.NoteFieldAccess(field,
+                                    read: value == null || WriteConsideredUse(fieldAccess.FieldSymbol.Type, value),
+                                    write: field.RefKind == RefKind.None || isRef);
```

Comment:
Sounds good, I can add `isRef = false`. I don't think I can assert anything useful because the receiver can be a ref field in an error scenario (I will add a test).

Reply:
Good catch, I think `isRef = false` should be there. Even though it would be an error scenario today to have a ref field to a ref struct, it could be allowed in the future, and the warnings can be observed even today. And if we are ref-assigning (i.e., `isRef = true`) we don't want to propagate that to the receiver (we are not ref assigning the receiver).